src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -829,10 +830,11 @@ void applyCompoundAssign(JCTree.JCExpression lhs, Function<Value, Value> scanRhs
                     FieldDesc fd = symbolToFieldDesc(sym, assign.selected.type);
 
                     Op.Result lhsOpValue;
+                    TypeDesc resultType = typeToDesc(sym.type); // why not sym.type
```

Comment:
This comment can be deleted. Before, I was getting the type using assign.selected.type, now sym.type seems to give the same result.